### DISCUSSION TITLE
Build prerendering page TypeError

### DISCUSSION THREAD

**Discussion by mvellandi** [vercel/next.js#37844] [Help] [answered]
Newcomer here. I can't figure out the following build prerender error:

`Error occurred prerendering page "/posts/[slug]". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of undefined (reading 'title')
    at Post (/.next/server/pages/posts/[slug].js:107:22)`

I reviewed the mentioned Next error docs, and tried adding an initial value for 'title' to null, to no avail.
Here's the mentioned page:
https://github.com/mvellandi/mv-travel-blog/blob/main/pages/posts/%5Bslug%5D.js

Any ideas? I thought with initial data fetching in getStaticProps, this wouldn't be a problem. Console logs are giving me complete component data props.
Thank you!

**Comment by icyJoseph** [comment, marked as answer]
Hi,

You are using `fallback: true` which means the server needs to pre-render a fallback page as well, where `posts` will be `undefined`!

This is done to account for the case where someone arrives to a page not yet generated, then a fallback is shown while the page is generated. Using `fallback: 'blocking'` doesn't cause this, but when you want to use `fallback: true`, you have to check if the router is in fallback mode:


```js
const Post = ({ post }) => {
  const {isFallback} = useRouter();

  if(isFallback) return <div>Loading...</div>
  
  const {
    title = null,
    categories,
    body,
    authorImage,
    username,
    about,
    postedAt,
  } = post;
```

You can see [here](https://github.com/icyJoseph/next-js-react-gbg/blob/main/demo-app/src/pages/pokemon/%5Bid%5D.tsx), how I do it for an application that generated as many Pokemon as needed, with fallback set to true.

This is well [documented on this page of the docs](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-pages).

**Reply by mvellandi** [reply]
Fallback situation indeed! Makes sense. Thanks so much!!!
